File: java/smartlist/src/main/java/com/smartlist/module/candidate/questionstrategy/TextStrategy.java
package com.smartlist.module.candidate.questionstrategy;

import com.smartlist.model.Question;
import com.smartlist.module.candidate.CandidateResponseCodes;
import com.smartlist.module.candidate.dto.request.SubmitAnswerReqDTO;
import com.smartlist.module.question.AnswerOptionRepository;
import com.smartlist.module.question.QuestionResponseCodes;
import org.apache.commons.lang3.StringUtils;

public class TextStrategy extends BaseQuestionStrategy{

    public TextStrategy(final SubmitAnswerReqDTO submitAnswerReqDto, final Question question,
                        final AnswerOptionRepository answerOptionRepository){
        super(submitAnswerReqDto, question, answerOptionRepository);

    }

    public void validate() {
        if (!this.submitAnswerReqDto.getAnswerIds().isEmpty()) {
            throw CandidateResponseCodes.INVALID_ANSWER_ID;
        }

        if (StringUtils.isEmpty(this.submitAnswerReqDto.getAnswerText())) {
            throw QuestionResponseCodes.INVALID_ANSWER_TEXT;
        }
    }

    @Override
    public int score() {
        return 0;
    }
}

File: javascript/smartlist/smartlist-api-mongodb-project/src/modules/candidate/candidate-question-strategy/text-strategy.ts
import { Injectable } from '@nestjs/common';
import { QuestionDocumentRepository } from 'src/modules/question/question.document.repository';
import { QuestionResponseCodes } from 'src/modules/question/question.response.codes';
import { QuestionDocument } from 'src/schemas/question.schema';
import { CandidateResponseCodes } from '../candidate.response.codes';
import { SubmitAnswerReqDto } from '../dto/request/submit-answer.req.dto';
import { BaseQuestionStrategy } from './base-question-strategy';

@Injectable()
export class TextStrategy extends BaseQuestionStrategy {
  constructor(submitAnswerReqDto: SubmitAnswerReqDto,
    questionDocument: QuestionDocument,
    questionDocumentRepository: QuestionDocumentRepository) {
    super(submitAnswerReqDto, questionDocument, questionDocumentRepository);
  }

  async validate() {
    if (!this.submitAnswerReqDto.answer_text) {
      throw CandidateResponseCodes.INVALID_ANSWER_TEXT;
    }
  }

  score(): number {
    return 0;
  }
}
